x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClustersEnrichIT.java

```diff
@@ -61,6 +61,8 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 
+// MP TODO: what to do about missing enrich policies on skip_unavailable=true clusters? Should that be fatal or non-fatal?
```

Comment:
I'd say they'd be SKIPPED too. If we are skipping on missing index, it makes sense treat the enrich policy the same way, especially given the potential move from enrich policies to indices in the future.

Reply:
After looking at this more closely, I don't see how to cleanly/easily handle missing enrich policies from skip_unavailable=true clusters. We would need to add knowledge to the Analyzer and LogicalPlanner about whether to not have that be a fatal error and then actively rewrite the users query to remove the offending cluster from the query entirely.

That's certainly beyond what I want to do in this PR and possibly not something we ever want to do. Open to feedback on it from the ES|QL team.

Reply:
Scratch that. I believe I've found a way to deal with missing enrich policies for skip_unavailable=true clusters without having to modify code in/around the LogicalPlan. Need to do some extensive testing on it, so that can either be a later commit to this PR or a follow-on PR.

Reply:
If I understood it correctly (very well could be not) the enrich resolution step is supposed to prepare a map where for each cluster we have a mapping that maps the policy to specific enrich indices, etc. Now, if we encounter the situation where, in remote enrich (in local-side enrich the local one should be used, as I understand) the policy is missing, and we mark the cluster as skipped, then nothing will be run on that cluster. So the missing map entry for this cluster should be ok? I _think_ the planner should be ok with that too, but we will need to ensure the verifier wouldn't give any trouble for skipped clusters. I'm not sure why query rewriting is necessary, would be glad to sync on this, I may be missing something.